src/ray/core_worker/core_worker.cc

```diff
@@ -260,6 +295,14 @@ void CoreWorker::ReportActiveObjectIDs() {
   heartbeat_timer_.async_wait(boost::bind(&CoreWorker::ReportActiveObjectIDs, this));
 }
 
+void CoreWorker::PromoteObjectToPlasma(const ObjectID &object_id) {
+  RAY_CHECK(object_id.IsDirectCallType());
+  auto value = memory_store_->GetOrPromoteToPlasma(object_id);
+  if (value != nullptr) {
+    plasma_store_provider_->Put(*value, object_id);
+  }
```

Comment:
Does it make sense to also update the in-memory store value of this to be the plasma error instead of the object value? Also, maybe it makes more sense to put the `plasma_store_provider_->Put` inside the memory_store_ method. It's a bit weird that there are two different ways to put objects into plasma.

Reply:
The call here is weird because I'm reusing the GetOrPromote call, which is also used elsewhere in a way that does require the return value.